src/main/java/com/karimbouchareb/chromatographysimulator/ui/SplashScreen.java

```diff
@@ -54,8 +60,9 @@ public void initSplashScreen(){
     /**
      * Initialize the rotating background image
      */
-    private ImageView initBackground(){
-        ImageView backgroundScatter = ChromatographySimulatorApp.makeImageView("backgroundPopulator.gif");
+    private ImageView initBackground() throws FileNotFoundException {
+        ImageView backgroundScatter = ChromatographySimulatorApp.makeImageView(BACKGROUND_SCATTER_FILE);
+        if (backgroundScatter == null) throw new FileNotFoundException("No file named " + BACKGROUND_SCATTER_FILE);
```

Comment:
Nice, I actually meant to tell you this to use the constant in forming the error message, but you beat me to it. Good coding spidey-sense.

Reply:
Also this error message is a lot better. You can take it one more step by naming the descriptions a bit more specific. Right now, you have three error messages that say "No file named " + foo. You can make it more descriptive if you say: "No file named " + foo + " when trying to initialize the background image for the splash screen" or something like that. 

Basically the goal is: if this string is the only thing an oncall engineer sees at 2am, can you give them an idea of what is wrong? Or will you make them have to dig to figure out the context? Because that engineer will likely be you.

Reply:
HAHA! @ "that engineer will likely be you"